File: solutions/0009-palindrome-number/solution.cpp
class Solution {
public:
    bool isPalindrome(int x) {
            int origx = x;
            if(x<0)
            {
                return false;
            }
            if(x<10)
            {
                return true;
            }
            if(x>2147483647)
            {
                return false;
            }
            int y = 10;
            int z = 1;
            int w = 0;
            int p = 1;
            while (x > 0)
            {
                p = x/y;
                z = x % y;
                if (w > (INT_MAX/10))
                    return false;
                w = w * 10 + z; 
                x = p;
            }
            return w == origx;
    }
};

File: solutions/0009-palindrome-number/solution.go
/*
 * Given an integer x, return true if x is a  palindrome, and false otherwise.
 */
func isPalindrome(x int) bool {
    const MaxUint = ^uint(0) 
    const MinUint = 0 
    const MaxInt = int(MaxUint >> 1) 
    const MinInt = -MaxInt - 1
    if x == 0 {
        return true
    }
    if x < 0 {
        return false
    }
    if x > 0 && x < 10 {
        return true
    }
    if x < 0 {
        x = int(math.Abs(float64(x)))
    }
    origX := x
    y := 10
    r := 0
    rev := 0
    q := 0
    for x != 0 {
        q = x / y
        r = x % y
        rev = rev * 10 + r
        x = q
    }
    return rev == origX
}
